### PR TITLE
Clean up `Runner`

### PR DESCRIPTION
**Pull request by desertaxle**
Cleans up the interface for `Runner`, removes dangling references to infrastructure, cleans up loop service cancellation in `Runner.flow_run_execute()`, renames `Runner.load` to `Runner.add`, and adds docstrings.

### CODE REVIEW

File: src/prefect/flows.py [hunk 1]
```diff
@@ -569,6 +573,14 @@ def my_flow(name):
             tags=tags,
             version=version,
         )
+        if print_starting_message:
+            console = Console()
```
Comment (cicdw) [inline src/prefect/flows.py:577]: interesting - what's the idea of using a `Console` object instead of a raw `print` statement?

Comment (cicdw) [reply src/prefect/flows.py:577]: Oh I see - is it for colored output?

File: src/prefect/runner.py [hunk 2]
```diff
@@ -105,54 +68,241 @@ def __init__(
...
+
+                    pid = await self._runs_task_group.start(
+                        self._submit_run_and_capture_errors, flow_run
+                    )
+
+                    self._flow_run_process_map[flow_run.id] = pid
+
+                    workload = partial(
+                        self._check_for_cancelled_flow_runs,
+                        on_nothing_to_watch=tg.cancel_scope.cancel,
```
Comment (cicdw) [inline src/prefect/runner.py:267]: whoa this looks great, nicely done

File: src/prefect/runner.py [hunk 3]
```diff
@@ -164,11 +314,15 @@ async def run(
         _use_threaded_child_watcher()
         flow_run_logger.info("Opening process...")
 
+        env = get_current_settings().to_environment_variables(exclude_unset=True)
+        env.update({"PREFECT__FLOW_RUN_ID": flow_run.id.hex})
+        env.update(**os.environ)  # is this really necessary??
```
Comment (cicdw) [inline src/prefect/runner.py:319]: nice i like bringing this closer to the process creation

**Review by cicdw — APPROVED** [review]
one question about the use of `Console` but otherwise this looks great

File: src/prefect/flows.py (hunk 1 above)
Comment (desertaxle) [reply src/prefect/flows.py:577]: Yup, it's for the looks. I think eventually we'll want to display a link to the UI and/or a CLI command to trigger a run, so the current output is more of a placeholder right now.

**Pull request merged**